From 542b999f8c3fe08ff080380bd715c05148b3d2cd Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 18:58:51 -0400
Subject: [fs] btrfs: add missing extent state caching calls

Message-id: <1407179109-13714-45-git-send-email-sandeen@redhat.com>
Patchwork-id: 87381
O-Subject: [RHEL7.1 PATCH 044/422] Btrfs: add missing extent state caching calls
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Filipe David Borba Manana <fdmanana@gmail.com>

commit c42ac0bc9530d51029b938e09b60b5ee86e5ee70
Author: Filipe David Borba Manana <fdmanana@gmail.com>
Date:   Tue Nov 26 15:01:34 2013 +0000

    Btrfs: add missing extent state caching calls

    When we didn't find a matching extent state, we inserted a new one
    but didn't cache it in the **cached_state parameter, which makes a
    subsequent call do a tree lookup to get it.

    Signed-off-by: Filipe David Borba Manana <fdmanana@gmail.com>
    Signed-off-by: Josef Bacik <jbacik@fb.com>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/extent_io.c b/fs/btrfs/extent_io.c
index 3ec5f05..6a68112 100644
--- a/fs/btrfs/extent_io.c
+++ b/fs/btrfs/extent_io.c
@@ -817,6 +817,7 @@ again:
   if (err)
    extent_io_tree_panic(tree, err);
 
+  cache_state(prealloc, cached_state);
   prealloc = NULL;
   goto out;
  }
@@ -1040,9 +1041,10 @@ again:
    goto out;
   }
   err = insert_state(tree, prealloc, start, end, &bits);
-  prealloc = NULL;
   if (err)
    extent_io_tree_panic(tree, err);
+  cache_state(prealloc, cached_state);
+  prealloc = NULL;
   goto out;
  }
  state = rb_entry(node, struct extent_state, rb_node);
-- 
1.7.1